test/state/block.hpp

```diff
@@ -18,15 +18,16 @@ struct Ommer
 
 struct Withdrawal
 {
-    uint64_t index = 0;
-    uint64_t validator_index = 0;
+    std::optional<uint64_t> index;
```

Comment:
I don't like this change because it mixes the data de-serialization error with a logic error. I think, the `Withdrawal` struct represent the correctly de-serialized object. Otherwise, I cannot expose such API do end users (and I plan to). Is there any better option handle this?